hilo.rb

```diff
@@ -8,17 +8,38 @@ def initialize
   end
 
   def play
+    previous_guesses = []
+
     while @guesses.positive?
       puts 'Guess the number between 1 & 100 (inclusive):'
-      guess = gets.chomp.to_i
+      guess = nil
+
+      loop do
+        input = gets.chomp
+        if input.match?(/^\d+$/)
+          guess = input.to_i
+          if guess.between?(1, 100)
+            if previous_guesses.include?(guess)
+              puts 'You have already guessed that number, try a different number:'
+            else
+              previous_guesses << guess
+              break
```

Comment:
As a general rule if you can avoid using `break` then do it. `break` makes it harder to figure out the exit conditions of a loop.